### PR TITLE
Davecaos.3.implement basic func

### PR DESCRIPTION
**Pull request by davecaos**
[#3] Basic implementation with dummy example of Common Test for Conferl module.

### CODE REVIEW

File: test/conferl_content_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,89 @@
+% This file is licensed to you under the Apache License,
```
Comment (gadgetci) [inline test/conferl_content_SUITE.erl]: According to **Elvis**:
> The module "conferl_content_SUITE" does not respect the format defined by the regular expression '"^([a-z][a-z0-9]*_?)*$"'.

File: src/conferl_content.erl [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Public Api
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%
+%% @doc functions definitions for content
+
+-spec register_content(Url :: iodata()) -> 
+  conferl_contents:content() | error.
+
+register_content(Url) -> #{}.
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> variable 'Url' is unused

File: src/conferl_content.erl [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+%% todo
+
+-spec unregister_content(ContentId :: integer()) -> ok | error .
+
+unregister_content(ContentId ) ->  error .
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> variable 'ContentId' is unused

File: src/conferl_content.erl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+%% todo
+
+-spec fetch_content(ContentId :: integer()) -> 
+   notfound | conferl_contents:content().
+
+fetch_content(ContentId) ->   notfound.   
+%% todo
+
+-spec list_contents(Domain :: iodata()) -> [conferl_contents:content()].
+list_contents(Domain) -> [ #{} ].
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> variable 'Domain' is unused

File: src/conferl_content.erl [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+-spec unregister_content(ContentId :: integer()) -> ok | error .
+
+unregister_content(ContentId ) ->  error .
+%% todo
+
+-spec fetch_content(ContentId :: integer()) -> 
+   notfound | conferl_contents:content().
+
+fetch_content(ContentId) ->   notfound.
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> variable 'ContentId' is unused

File: test/conferl_content_SUITE.erl (hunk 1 above)
Comment (elbrujohalcon) [inline test/conferl_content_SUITE.erl]: @davecaos you need to update your elvis.config

File: src/conferl_content.erl [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+-spec sumo_wakeup(sumo:doc()) -> content().
+sumo_wakeup(Data) ->  Data.
+
+%% @doc Part of the sumo_doc behavior.
+-spec sumo_sleep(content()) -> sumo:doc().
+sumo_sleep(Content) ->  Content.
+
+%% @doc Part of the sumo_doc behavior.
+-spec sumo_schema() -> sumo:schema().
+sumo_schema() ->
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> function sumo_schema/0 is unused

File: src/conferl_content.erl [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+         }.
+
+-export_type( [content/0]).
+
+-export([ register_content/1
+          , unregister_content/1
+          , fetch_content/1
+          , list_contents/1 ]).
+
+-behavior(sumo_doc).
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> undefined callback function sumo_schema/0 (behaviour 'sumo_doc')

File: src/conferl_content.erl [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% BEHAVIOUR CALLBACKS
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% @doc Part of the sumo_doc behavior.
+-spec sumo_wakeup(sumo:doc()) -> content().
+sumo_wakeup(Data) ->  Data.
+
+%% @doc Part of the sumo_doc behavior.
+-spec sumo_sleep(content()) -> sumo:doc().
+sumo_sleep(Content) ->  Content.
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> function sumo_sleep/1 is unused

File: src/conferl_content.erl (hunk 7 above)
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> undefined callback function sumo_sleep/1 (behaviour 'sumo_doc')

File: src/conferl_content.erl [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+-spec list_contents(Domain :: iodata()) -> [conferl_contents:content()].
+list_contents(Domain) -> [ #{} ].
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% BEHAVIOUR CALLBACKS
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% @doc Part of the sumo_doc behavior.
+-spec sumo_wakeup(sumo:doc()) -> content().
+sumo_wakeup(Data) ->  Data.
```
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> function sumo_wakeup/1 is unused

File: src/conferl_content.erl (hunk 7 above)
Comment (gadgetci) [inline src/conferl_content.erl]: According to **Compiler**:
> undefined callback function sumo_wakeup/1 (behaviour 'sumo_doc')

**Comment by davecaos** [comment]
Closed.

**Pull request closed without merging**